sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -81,6 +86,19 @@ private[spark] class HiveExternalCatalog(client: HiveClient, hadoopConf: Configu
     withClient { getTable(db, table) }
   }
 
+  /**
+   * If the given table properties contains datasource properties, throw an exception.
+   */
+  private def verifyTableProperties(table: CatalogTable): Unit = {
+    val datasourceKeys = (table.properties.keys ++ table.storage.properties.keys)
+      .filter(_.startsWith(DATASOURCE_PREFIX))
+    if (datasourceKeys.nonEmpty) {
+      throw new AnalysisException(s"Cannot persistent ${table.qualifiedName} into hive metastore " +
+        s"as table/storage property keys may not start with '$DATASOURCE_PREFIX': " +
+        datasourceKeys.mkString("[", ", ", "]"))
+    }
+  }
```

Comment:
actually, I am wondering if we should keep it in the properties? So, it is easy for us to debug?